crates/astria-composer/src/main.rs

```diff
@@ -1,29 +1,42 @@
+use std::process::ExitCode;
+
 use astria_composer::{
     telemetry,
     Composer,
     Config,
 };
+use color_eyre::eyre::WrapErr as _;
 use tracing::info;
 
 #[tokio::main]
-async fn main() {
+async fn main() -> ExitCode {
     let cfg: Config = match config::get() {
         Ok(cfg) => cfg,
         Err(e) => {
             eprintln!("failed to read configuration: {e}");
-            std::process::exit(2);
+            return ExitCode::FAILURE;
         }
     };
+
+    if let Err(e) = telemetry::configure()
+        .otel_endpoint("http://otel-collector.monitoring:4317")
+        .filter_directives(&cfg.log)
+        .try_init()
+        .wrap_err("failed to setup telemetry")
+    {
+        eprintln!("initializing sequencer failed:\n{e:?}");
+        return ExitCode::FAILURE;
+    }
```

Comment:
is the otel-collector something run locally or remotely? will starting up the service fail is the otel-collector is not available?

Reply:
After your comment I questioned my decision to call this a non-breaking change. I added config options optionally disable OTLP so that this can be run locally without an otel collector.